zebROS_ws/src/behaviors/src/2023_fourber_server.cpp

```diff
@@ -0,0 +1,383 @@
+      // rememeber where we were when moving ot saftey mode as the only one that uses saftey mode is the elevatER
+      previous_setpoint_ = fourbar_cur_setpoint_;
+      if (goal->safety_position == fourber_ns::SAFETY_HIGH) {
+        FourberINFO("Safey HIGH mode called for fourbar");
+        saftey_state_ = SafteyState::SAFTEY_HIGH;
+        safetyBoundsAndCallService(safety_high_distance_, safety_high_below_);
+      }
+
+      if (goal->safety_position == fourber_ns::SAFETY_INTAKE_LOW) {
+        FourberINFO("Safey LOW mode called for fourbar");
+        saftey_state_ = SafteyState::SAFTEY_LOW;
+        safetyBoundsAndCallService(safety_low_distance_, safety_low_below_); 
+      }
+      return;
+    }
+
+    // case where we have move back to where we were
+    if (goal->safety_position == fourber_ns::SAFETY_TO_NO_SAFETY) {
+      saftey_state_ == SafteyState::NONE;
+      controllers_2023_msgs::FourBarSrv go_to_previous_req;
+      go_to_previous_req.request.position = previous_setpoint_;
+      // should be good to hardcode this because the only time we want this to be true is when we are intaking which is in a restricted zone
+      go_to_previous_req.request.below = false; 
+
+      if (!fourbar_srv_.call(go_to_previous_req)) {
+        FourberERR("Failed calling fourber with message 'go_to_previous_req'");
+        publishFailure();
+      }
+
+      if (!waitForFourbar(go_to_previous_req.request.position)) {
+        publishFailure();
+      }
+
+      publishSuccess(); 
+      return;
+    }
+
+    // select piece, nice synatax makes loading params worth it
+    double req_position = game_piece_lookup_[goal->piece][goal->mode].first;
+    bool req_bool = game_piece_lookup_[goal->piece][goal->mode].second;
+
+    // apply offset
+    req_position += position_offset_;
+    if (!position_offset_) {
+      FourberWARN("Offset of " << position_offset_);
+    }
+
+    // for movements when inside a safe zone
+    if (saftey_state_ == SafteyState::SAFTEY_HIGH) {
+      req_position = std::max(req_position, safety_high_distance_);
+      req_bool = safety_high_below_;
+    }
+    if (saftey_state_ == SafteyState::SAFTEY_LOW) {
+      req_position = std::max(req_position, safety_low_distance_);
+      req_bool = safety_low_below_;
+    }
+
+    FourberINFO("FourbERing a " << piece_to_string[goal->piece] << " to the position " << mode_to_string[goal->mode] << " and the FOURBAR to the position=" << req_position << " meters");
+
+    // we know that saftey is set to none 
+    controllers_2023_msgs::FourBarSrv req;
+    req.request.position = req_position;
+    req.request.below = req_bool;
+
+    if (!fourbar_srv_.call(req)) { // somehow fourbar has failed, set status and abort to pass error up
+      FourberERR("Failed to moving fourbar :(");
+      publishFailure();
+      return;
+    }
+
+    // failed 
+    if (!waitForFourbar(req_position)) {
+      publishFailure();
+    }
```

Comment:
Probably need a return in this if case, otherwise it will fall through and publish success